R/corPFSOS.R

```diff
@@ -0,0 +1,123 @@
+# E(PFS) helpers:
+survPFS <- function(transition, t) {
+  UseMethod("survPFS")
+}
+
+survPFS.ExponentialTransition <- function(transition, t) {
+  ExpSurvPFS(t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02)
+}
+
+survPFS.WeibullTransition <- function(transition, t) {
+  WeibSurvPFS(
+    t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    p01 = transition$weibull_rates$p01, p02 = transition$weibull_rates$p02
+  )
+}
+
+survPFS.PWCTransition <- function(transition, t) {
+  PWCsurvPFS(t,
+    h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    pw01 = transition$intervals$pw01, pw02 = transition$intervals$pw02
+  )
+}
+
+# E(OS) helpers:
+survOS <- function(transition, t) {
+  UseMethod("survOS")
+}
+
+survOS.ExponentialTransition <- function(transition, t) {
+  ExpSurvOS(
+    t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    h12 = transition$hazards$h12
+  )
+}
+
+survOS.WeibullTransition <- function(transition, t) {
+  WeibSurvOS(
+    t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    h1 = transition$hazards$h1, p01 = transition$weibull_rates$p01,
+    p02 = transition$weibull_rates$p02, p12 = transition$weibull_rates$p12
+  )
+}
+
+survOS.PWCTransition <- function(transition, t) {
+  PWCsurvOS(
+    t = t, h01 = transition$hazards$h01, h02 = transition$hazards$h02,
+    h12 = transition$hazards$h12, pw01 = transition$intervals$pw01,
+    pw02 = transition$intervals$pw02, pw12 = transition$intervals$pw12
+  )
+}
+
+# Var(PFS)/Var(OS) helpers:
+expvalPFSInteg <- function(transition, t) {
+  t * survPFS(transition, t)
+}
+
+expvalOSInteg <- function(transition, t) {
+  t * survOS(transition, t)
+}
+
+# E(PFS*OS) helpers:
+
+#------- P_11 (see Meller et al, 4274)
+
+p11Integ <- function(x, transition) {
+  haz(transition = transition, t = x, trans = 3)
+}
+
+p11 <- function(transition, s, t) {
+  intval <- mapply(function(s, t) integrate(p11Integ, lower = s, upper = t, transition)$value, s, t)
+  exp(-intval)
```

Comment:
I notice the `exp()` here and the multiplication in line 77 - any chance to do line 77 on the log scale as an addition first and then `exp()` the result?

Reply:
I have tried it, integrating over log(a) + log(b) + log(c) and then doing exp() returns non-finite function values in some cases, leaving it like this does not.

Reply:
I would not do the integration over the logs, just calculate the integrand via exp(log(a) + log(b) + log(c)) instead of a * b * c - but not sure if we have log options for survPFS and haz functions?

Reply:
1. I see what you mean. What would be the advantage for doing it like this?
2. What do you mean with _log option_?